pkg/collector/corechecks/servicediscovery/usm/service.go

```diff
@@ -219,7 +227,7 @@ func ExtractServiceMetadata(args []string, envs map[string]string, fs fs.SubFS,
 
 	exe = normalizeExeName(exe)
```

Comment:
IIRC, the exe is actually `argv[0]`, so you can still have `python3.10`
also, here we don't support `py3` for instance

I think we should get the language information (if possible) and try to detect service name by that

Reply:
I agree. But I think that should be done in a separate PR, and maybe have this current method (and this fix to it) as a fallback is language detection fails?